.github/actions/flatpak-builder-lint/action.yaml

```diff
@@ -0,0 +1,63 @@
+name: Run flatpak-builder-lint
+description: Runs flatpak-builder-lint with exceptions
+inputs:
+  artifact:
+    description: Type of artifact to lint (builddir, repo, manifest)
+    required: true
+  path:
+    description: Path to flatpak-builder manifest or Flatpak build directory
+    required: true
+  validateToPublish:
+    description: If false, turns some errors to non-errors for non-publish workflow
+    required: false
+    default: true
+  workingDirectory:
+    description: Working directory to clone flatpak-builder-lint
+    required: false
+    default: ${{ github.workspace }}
+runs:
+  using: composite
+  steps:
+    - uses: actions/checkout@v3
+      with:
+        repository: flathub/flatpak-builder-lint
+        ref: v2.0.13
+        path: flatpak-builder-lint
+        set-safe-directory: ${{ inputs.workingDirectory }}
+
+    - name: Install Dependencies 🛍️
+      shell: bash
+      working-directory: ${{ inputs.workingDirectory }}
+      run: |
+        : Install Dependencies 🛍️
+        echo ::group::Install Dependencies
+        dnf install -y -q poetry jq
+        poetry -q -C flatpak-builder-lint install
+        echo ::endgroup::
+
+    - name: Run flatpak-builder-lint
+      id: result
+      shell: bash
+      working-directory: ${{ inputs.workingDirectory }}
+      run: |
+        : Run flatpak-builder-lint
+        ret=$(poetry -C flatpak-builder-lint run flatpak-builder-lint --exceptions ${{ inputs.artifact }} ${{ inputs.path }} || true)
+
+        for ((i = 0 ; i < $(echo $ret | jq '.warnings | length') ; i++)); do
```

Comment:
Instead of using counters (and then run `jq` on the returned data again), just loop over the entries returned by `jq` - if no values are returned, no loop iteration runs:

```suggestion
            local success=1
            poetry -C flatpak-builder-lint run flatpak-builder-lint --exceptions ${{ inputs.artifact }} ${{ inputs.path }} > linter_output.json || success=0

            if (( success )); then
              while read -r warning; do
                  echo "::warning::${warning} found in the Flatpak ${{ inputs.artifact }}"
              done <<< "$(jq -r '.warnings | .[]?' linter_output.json )"
```
The `jq` expression will only yield warning entries if the key actually exists, otherwise the output is empty (and return code is always `0`). As each warning entry will be printed to stdout on a single line, `while` will automatically iterate over each warning.

Potentially even more elegant would be to handle all that in a single loop:

```
              IFS=';' while read -r type warning; do
                  if [[ "${type}" == "warning" ]]; then
                    echo "::warning::${warning} found in the Flatpak ${{ inputs.artifact }}"
                  elif [[ "${type}" == "error" && "${{ inputs.validateToPublish }}" == "true" ]]; then
                    echo "::error::${warning} found in the Flatpak ${{ inputs.artifact }}"
                  fi
              done <<< "$(jq -r '(.warnings | try map("warning," + .)) + (.errors | try map("error," + .)) | .[]?' linter_output.json)"
```

Reply:
- When flatpak-builder-lint find an error in the manifest/builddir/repo, it returns a non-zero exit code
- The quote are kept on purpose in the annotations
- Notice are added to show "ignored" without marking them as red errors
- `validateToPublish` does not ignore everything just specific error that are clearly fixed in the publish CI
- I purposefully do not write files to not create unneeded files

Reply:
Using while introduce an empty line when the linter don't find any issue (empty variable/file).

```
while read -r warning; do
echo "::warning::$warning found in the Flatpak manifest"
done <<< "$(jq '.warnings[]?' test.json)"

::warning:: found in the Flatpak manifest
```

I'll keep for loops for now.